mako/lexer.py

```diff
@@ -357,10 +357,12 @@ def match_text(self):
             r"""
                 (.*?)         # anything, followed by:
                 (
-                 (?<=\n)(?=[ \t]*(?=%|\#\#)) # an eval or line-based
+                 ((?<=\n)|^)(?=[ \t]*(?=%|\#\#)) # an eval or line-based
```

Comment:
i dont think the `|^` is needed

Reply:
also we can remove the "%" from this line

Reply:
Can't remove the "%" here, doing so would cause the regex in `match_text` failed to consume the newline character ("\n") such as `"\n %for"`. Removal would result in the `match_control_line`'s regular expression failing to recognize these control keywords.
https://github.com/sqlalchemy/mako/blob/dc66614a2b9ba19ee880c855471c37cd3307c22a/mako/lexer.py#L424-L429